indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerMapper.java

```
@@ -109,4 +153,37 @@ public final static InputRow parseInputRow(Object value, InputRowParser parser)
   abstract protected void innerMap(InputRow inputRow, Object value, Context context, boolean reportParseExceptions)
       throws IOException, InterruptedException;
 
+  private List<String> mergedDimensions = null;
+
+  private InputRow mergePartitionDimValues(InputRow inputRow)
+  {
+    InputRow merged = inputRow;
+
+    // only for raw data case
+    if (inputRow instanceof MapBasedInputRow) {
```

can we throw an exception if not supported?